### PR TITLE
CODENVY-1711: Grant permissions to admins for predefined stacks

### PR DESCRIPTION
**Pull request by mmorhun**
### What does this PR do?
Allows to users with `manageSystem` permission perform edit and delete operation with predefined (i.e. public for all users) stack.

### What issues does this PR fix or reference?
https://github.com/codenvy/codenvy/issues/1711

#### Changelog
Permissions granted to admins to perform `update` and `delete` action with predefined stacks.

#### Release Notes
Added ability to admins with `manageSystem` permission perform `update` and `delete` action with predefined stacks.

#### Docs PR
N/A

### CODE REVIEW

File: wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/StackPermissionsFilter.java [hunk 1]
```diff
@@ -86,8 +103,66 @@ public void filter(GenericResourceMethod genericResourceMethod, Object[] argumen
                 throw new ForbiddenException("The user does not have permission to perform this operation");
         }
 
+        if (currentSubject.hasPermission(SystemDomain.DOMAIN_ID, stackId, SystemDomain.MANAGE_SYSTEM_ACTION)
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/StackPermissionsFilter.java]: currentSubject.hasPermission(SystemDomain.DOMAIN_ID, **null**, SystemDomain.MANAGE_SYSTEM_ACTION)

File: wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/StackPermissionsFilter.java [hunk 2]
```diff
@@ -86,8 +103,66 @@ public void filter(GenericResourceMethod genericResourceMethod, Object[] argumen
...
+    private Page<AbstractPermissions> getNextPermissionsPage(String stackId,
+                                                             @Nullable Page<AbstractPermissions> permissionsPage) throws NotFoundException,
+                                                                                                                         ConflictException,
+                                                                                                                         ServerException {
+        if (permissionsPage == null) {
+            // get first page
+            return permissionsManager.getByInstance(DOMAIN_ID, stackId, 25, 0);
+        }
+        if (!permissionsPage.hasNextPage()) {
+            return null;
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/StackPermissionsFilter.java]: I don't think that it is good idea to return null here

**Comment by mmorhun** [comment]
ci-build

**Comment by sleshchenko** [comment]
Add yourself as author to StackPermissionsFilter please

**Comment by bmicklea** [comment]
These will appear in Swagger docs, right?

**Comment by mmorhun** [comment]
@bmicklea no, it will just perform action instead of error with permission denied message in case of predefined stack.

**Pull request merged**